src/utility/PlaybackService.ts

```diff
@@ -72,6 +82,24 @@ export default async function() {
                 sendPlaybackEvent('/Sessions/Playing/Stopped', settings.jellyfin);
             }
         }
+
+        // Handle is playback state is playing
+        if (event.state === State.Playing) {
+            const settings = store.getState().settings;
+            
+            // Start timer is sleeper is enabled
+            if (settings.enabledSleeper) {
+                let time = settings.sleepTime;
+                interval = setInterval(() => {
+                    if (time > 0) {
+                        time -= 1;
+                    } else {
+                        TrackPlayer.pause();
+                        clearInterval(interval);
+                    }
+                }, 1000);
+            }
+        }
```

Comment:
Rather than setting custom intervals, please save a `Date` in Redux when the timer is set to trigger, and then use the `Event.PlaybackProgressUpdated` event listener to check whether a timer is set and whether we're past the date.